core/tests/puppeteer-acceptance-tests/spec/exploration-creator-tests/savedraft-publish-and-discard-the-changes.spec.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for Exploration Creator and Exploration Manager
+ */
+
+import testConstants from '../../puppeteer-testing-utilities/test-constants';
+import { UserFactory } from '../../puppeteer-testing-utilities/user-factory';
+import { ExplorationCreator } from '../../user-utilities/exploration-creator-utils';
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+
+describe('Exploration Publisher, Saver and Drafter', function() {
+  let explorationCreator: ExplorationCreator;
+  let explorationVisitor: ExplorationCreator;
+  beforeAll(async function() {
+    explorationCreator = await UserFactory.createNewUser(
+      'explorationAdmi', 'exploration_creator@example.com');
+    explorationVisitor = await UserFactory.createNewUser(
+      'explorationVisitori', 'exploration_visitor@example.com');
```

Comment:
I think the usernames provided are off by one character.

Reply:
Actually need to change their name again and again due to datastore/cache so by mistake it remained like this. done ✅